src/client/Client.cc

```diff
@@ -624,7 +625,7 @@ void Client::trim_cache(bool trim_kernel_dcache)
     trim_dentry(dn);
   }
 
-  if (trim_kernel_dcache && lru.lru_get_size() > max)
+  if (trim_kernel_dcache && before > max)
```

Comment:
My understanding is that we only want to ask the kernel to invalidate its dentry cache **only if** the Client cannot shrink its cache by trimming unpinned items. It seems to me this check is already correct: if the LRU size is greater than the maximum client cache size then ask the kernel to drop its dentry cache for this mount (via remount).

cc @jtlayton

Reply:
https://github.com/ceph/ceph/blob/c32350ed98bea42d8c6911bcc5b26e1231c69c0a/src/client/Client.cc#L6238

Client::tick() passes **true** to **trim_kernel_dcache**, the kernel should drop its cache.  But it never did, and we can't always do this by remounting.

Reply:
I have the same question patrck had.  If trim_cache(true) succeeds in making cache size smaller than configured max size, why should it do the expensive remount to trim kernel cache

Reply:
max size: 16384
before = lru.lru_get_size = 16385
After trimming a dentry(lru.lru_get_size = 16384), it will not invalidate the kernel dcache.

Reply:
Which is what we'd want to happen, right? The LRU would be 16384 at that point which is the max. We didn't need to invalidate the kernel's dcache because we were able to get under the limit by invalidating dentries that the kernel already wasn't using.

Reply:
If this is what we expect, we can delete it. because the check condition will never be true.

Reply:
Ok, thanks. I'll go ahead and close this out.